[Lee Zen Been] iP

Zenlzb opened the pull request:
# Duke++
> "Quote blocks are very handy, and also try out Duke++" - Obama maybe

Duke++has many features you don't want to miss out on, such as:
* Super convenient!
* Very Responsive!
* _All-in-one_ task manager!
* Extremely **rad**! :ok_hand:
* Contains at least 1 `java`, like so

```java
@Override
    public String typeIcon() {
        return "[D]";
    }
}
```


How do you get your hands on such an amazing product? Simple!
1. Download it [here](https://github.com/Zenlzb/ip/releases/tag/v0.1)
2. Run it
3. Start Talking to Duke++!

There are many planned features as well!

- [x] Adding task
- [x] Deleting task
- [ ] Renaming task (_coming soon_)
- [ ] GUI (_coming soon_)
- [x] Being cool (~~_coming soon_~~ is cool)

Commit:
Add Gradle support

Commit:
Implement a Skeletal Version of Duke

Commit:
Add List and Add to List Functionality

Commit:
Add Task Completion Status and Complete Task Functionality

Commit:
Add Todo Task Functionality

Commit:
Add Deadline Task Functionality

Commit:
Minor Changes
- Change Syntax of Deadline command

Commit:
Add Event Task Functionality

Commit:
Restructure main body to use switch cases

Commit:
Add Text UI Testing

Commit:
Add Error Handling
- No arguments after done

Commit:
Add Delete Task Functionality and Update Test Cases

Commit:
Change Constructor Signature
- Includes Completion Status now

Commit:
Add Read and Write to file Functionality

Commit:
Implement Date-Time Parsing Functionality

Commit:
Merge and Resolve Merge Conflicts

Commit:
Bug Fix
- Date Format not working as intended

Commit:
Add Storage and TaskList classes
- Storage for handling reading and writing to file
- TaskList for handling operations on the task list

Commit:
Improve formatting of messages

Commit:
Add abstract class Command and implement first 2 commands ByeCommand and ListCommand

Commit:
Add Parser class to parse command input

Commit:
Update UI class to include more messages

Commit:
Begin restructuring of Duke class to use Command objects

Commit:
Divide all classes into packages, commands into sub-package

Commit:
Restructure getFormats method

Commit:
Add Formats and Todo Command and edit execute signature

Commit:
Add Task completed and deleted messages

Commit:
Add Delete and Done Commands

Commit:
Add Deadline and Delete Commands

Commit:
Fix Minor Bugs

Commit:
Complete restructure to use new parser and commands

Zenlzb committed:
Update README.md

Commit:
Minor change

Commit:
Add JUnit Tests

Commit:
Add Manifest

Commit:
Format code to follow Java Coding Standard

Commit:
Add Find Command

Commit:
Add JavaDoc to public classes and methods

Commit:
Merge branch-Level-9

Commit:
Merge branch-A-JavaDoc and resolve conflicts

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle
Add gradle support

Commit:
Add Gradle Support

Commit:
Add Gradle Support

Commit:
Add Checkstyle Support

Commit:
Merge branch 'branch-Level-9'

Commit:
Add Dependencies for JavaFX

Commit:
Refactor Commands to return a String upon execution

Commit:
Implement a basic GUI for interacting with Duke

Commit:
Restructure methods for interacting with GUI

wanninglim commented on src/main/java/duke/DateTimeHandler.java:
```diff
@@ -0,0 +1,77 @@
+    private final DateTimeFormatter[] dtfList = new DateTimeFormatter[formats.length];
+
+    public DateTimeHandler() {
+        for(int i=0; i<formats.length; i++) {
```
Could possibly have spaces before operators! Same goes for the rest of the code :)

Zenlzb replied:
Fixed

wanninglim commented on src/main/java/duke/DateTimeHandler.java:
```diff
@@ -0,0 +1,77 @@
+    private boolean tryToParse(DateTimeFormatter dtf, String s) {
+        try {
+            LocalDateTime.parse(s, dtf);
+        } catch(DateTimeParseException e) {
```
Spaces before parentheses would improve readability as well :)

Zenlzb replied:
Fixed

wanninglim commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,49 @@
+/**
+ * Encapsulates a class that contains methods for interacting with the user
+ */
+public class UI {
```
Instead of UI (full caps), it'll match coding standards more if the class is refactored to 'Ui' (pascal case)!

Zenlzb replied:
Fixed

wanninglim commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,49 @@
+     *
+     * @return the String message
+     */
+    public String welcomeMessage() {
```
`welcomeMessage()` is a noun! Could possibly change to `printWelcomeMessage()`, or prefix it with another action verb :)

Zenlzb replied:
Fixed

wanninglim commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,49 @@
+     *
+     * @return the String message
+     */
+    public String goodByeMessage() {
```
The same comment goes for this method name, and the one below it :D

Zenlzb replied:
Fixed

wanninglim commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,61 @@
+     *
+     * @return The command prefix
+     */
+    public abstract String startsWith();
```
Could possible change this to `getStartingPrefix()`/`getCommandPrefix()` to make it like a verb!

Zenlzb replied:
Changed to getCommandPrefix

wanninglim commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+                String[] parts = curLine.split(" ", 3);
+                String[] descriptionParts;
+                char taskType = parts[0].charAt(1);
+                boolean completionStatus = parts[1].charAt(1) == 'X';
```
`isCompleted` might make for a more understandable boolean variable!! :)

Zenlzb replied:
Fixed, Thanks for your help

wanninglim reviewed (changes requested):
Hello!

Generally LGTM, just need to take note of spaces before operations and change your method names to verbs for readability! :-)

Commit:
Change formats array into constant

Commit:
Change commandList array into constant

Commit:
Update JavaDocs

Commit:
Fix Styling issues raised by Checkstyle and Peer Review

Commit:
Add assertions to check that various assumptions hold

The code currently does not contain any assertions.

Assertions help document important assumptions and greatly help with testing.

Let's add assertions to various points in the code to make sure the assumptions are upheld.

Commit:
Change code to improve code quality

There are some lines of code which are of low quality.

Low quality code could lead to decreased readability or even cause bugs.

Let's change the low quality code so that the code will be more readable and less bug prone.

Zenlzb committed:
Merge pull request #1 from Zenlzb/branch-A-CodeQuality

Change code to improve code quality

Zenlzb committed:
Merge pull request #2 from Zenlzb/branch-A-Assertions

Add assertions to check that various assumptions hold

Commit:
Add Task tagging functionality

Tasks do not have the ability to be tagged.

Tagging would allow tasks to be searched easier while keeping description brief.

Let's add the ability to add tags to tasks.

Commit:
Fix Code Styling issues

alyssa-savier commented on src/main/java/duke/commands/TodoCommand.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Encapsulates the todo command, used to create a todo task
+ */
+public class TodoCommand extends Command {
```
I like your use of OOP.

alyssa-savier commented on src/main/java/duke/commands/FormatsCommand.java:
```diff
@@ -0,0 +1,31 @@
+import duke.TaskList;
+import duke.Ui;
+
+/**
```
I like the use of Java Docs.

alyssa-savier commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+     * @param t The Task that was deleted.
+     * @return The String array to be printed.
+     */
+    public String[] taskDeletedMessage(Task t) {
```
Does this follow the Java coding naming standard?

alyssa-savier commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return The String icon.
+     */
+    public String completionIcon() {
```
Again, is this in line with java naming standards?

alyssa-savier commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,73 @@
+    ____________________________________________________________
+     Hello! I'm duke.Duke
+     What can I do for you?
+    ____________________________________________________________
```
I like the use of textUi testing.

Commit:
Refactor tags from String Array to String ArrayList

Commit:
Add Tag Command to tag tasks after they have been created

Commit:
Add Styling to DialogBox

Commit:
Add Background Image

Commit:
Add Background Image

Commit:
Change DialogBox to adjust height with text

Commit:
Add Welcome Message

Commit:
Bug Fixes
- Fix DialogBox unintended scaling

Commit:
Remove unnecessary images

Commit:
Add DukeException handling

Commit:
Add DukeException handling

Commit:
Convert error messages to DukeException

Commit:
Convert error messages to DukeException

Commit:
Change Welcome Message

Commit:
Add JavaDoc and fix style errors

Commit:
Add more date formats

Commit:
Add Ui.png

Commit:
Fix Tests

Commit:
Fix styling problems

Commit:
Update build.gradle

Zenlzb committed:
Update User Guide

Zenlzb committed:
Set theme jekyll-theme-midnight

Zenlzb committed:
Update README.md
